src/element.ts

```diff
@@ -491,4 +488,95 @@ export abstract class Element {
     const fontInfo = this.fontInfo;
     this.setFont(fontInfo.family, fontInfo.size, weight, fontInfo.style);
   }
+
+  /** Get element width. */
+  getWidth(): number {
+    return this.width;
+  }
+
+  /** Set element width. */
+  setWidth(width: number): this {
+    this.width = width;
+    return this;
+  }
+
+  /** Shift element down `yShift` pixels. Negative values shift up. */
+  setYShift(yShift: number): this {
+    this.yShift = yShift;
+    return this;
+  }
+
+  /** Get shift element `yShift` */
+  getYShift(): number {
+    return this.yShift;
+  }
+
+  /**
+   * Set shift element right `xShift` pixels. Negative values shift left.
+   */
+  setXShift(xShift: number): this {
+    this.xShift = xShift;
+    return this;
+  }
+
+  /** Get shift element `xShift` */
+  getXShift(): number {
+    return this.xShift;
+  }
+
+  /**
+   * Set element text
+   */
+  setText(text: string): this {
+    this.text = text;
+    return this;
+  }
+
+  /** Get element text */
+  getText(): string {
+    return this.text;
+  }
+
+  /**
+   * Render the element text .
+   */
+  renderText(ctx: RenderContext, xPos: number, yPos: number): void {
+    ctx.save();
+    ctx.setFont(this.textFont);
+    ctx.fillText(this.text, xPos + this.xShift, yPos + this.yShift);
+    ctx.restore();
+  }
+
+  /** Canvas used to measure text */
+  static #txtCanvas?: HTMLCanvasElement;
+
+  measureText(): TextMetrics {
+    let txtCanvas = Element.#txtCanvas;
+    if (!txtCanvas) {
+      // Create the SVG text element that will be used to measure text in the event
+      // of a cache miss.
+      txtCanvas = document.createElement('canvas');
```

Comment:
Is the comment incorrect? Here you create a `canvas` element.

Will this approach work for all the environments that VexFlow 4 supports? In this case, we require the environment to be able to create a HTMLCanvasElement. What about NodeJS environments or [web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API)?

I know some other folks added support for [OffscreenCanvas](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas) to VexFlow, and there was node canvas.

Anyways, it is OK for VexFlow 5 to (temporarily) break support for those environments, but we should we aware that this approach (**requiring a HTMLCanvasElement for measurements**) is not a solution for everyone.

Reply:
Comment corrected. Do you know of any concrete example where this is not a solution? Why do you say that "this approach (requiring a HTMLCanvasElement for measurements) is not a solution for everyone".